File: 1422-maximum-score-after-splitting-a-string/solution.c
#include <string.h>


int maxScore(char* s) {
    int left_zeros = 0, right_ones = 0, max_score = 0;
    // count total number of ones
    for (int i = 0; s[i]!='\0'; i++) {
        if (s[i] == '1') right_ones++;

    }
    
    // traverse the string
    for (int i = 0; s[i+1] != '\0'; i++) {
        if (s[i] == '0'){
            left_zeros++;

        } else {
            right_ones--;
        }

        // calculate the score
        int score = left_zeros + right_ones;
        if (score > max_score) max_score = score;
    }

    return max_score;
}

File: 1422-maximum-score-after-splitting-a-string/solution.py
class Solution:
    def maxScore(self, s:str) -> int:
        # initialize the counts
        left_zeros = 0
        right_ones = s.count('1') # how many ones are in the string
        max_score = 0


        # now traverse the string
        for i in range(len(s)-1): # since split point must leave a non-empty substring
            if s[i] == '0':
                left_zeros += 1
            else:
                right_ones -= 1

            # now calculate the score
            max_score = max(max_score, left_zeros + right_ones)

        return max_score
